.github/workflows/ci.yaml

```diff
@@ -81,14 +91,48 @@ jobs:
   #unit-test:
   #runs-on: "ubuntu-latest"
 
-  setup-kubernetes-cluster:
+  k8s-cypress-tests:
     runs-on: "ubuntu-latest"
     steps:
-      - name: AbsaOSS/k3d-action
-        uses: AbsaOSS/k3d-action@v2.4.0
+      - uses: actions/checkout@v3
+      - uses: AbsaOSS/k3d-action@v2.4.0
         with:
           cluster-name: "Test-Cluster"
           args: --agents 1
 
+      - name: Prepare kubeconfig # Copy kubeconfig into /Proxy/.kube folder
+        run: |
+          cd Proxy && mkdir .kube && cd .kube
+          cp -r $HOME/.kube/config .
+          cat config
+
+      - name: Start kit
+        run: docker compose up -d
+      - run: sleep 20 # Wait for the proxy to collect some data
+
+      - name: Cypress run e2e testing
+        uses: cypress-io/github-action@v5
+        with:
+          working-directory: Explorer
+          headed: true
+        env:
+          CYPRESS_DEFAULT_COMMAND_TIMEOUT: 10000 # Set timeout to 10 seconds
+
+      - name: Upload screenshots
+        uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: cypress-snapshots
+          path: Explorer/cypress/screenshots
+          if-no-files-found: ignore
+          retention-days: 1
+
+      - name: Cypress run component testing
+        uses: cypress-io/github-action@v5
+        with:
+          working-directory: Explorer
+          component: true
+
+
```

Comment:
So when component tests fail we don't need screenshots?

Reply:
I just saw that the failing tests also produce a video, should we save that as well?

Reply:
already done